# Naive Bayes Classification - Diabetes Dataset

Tugas Machine Learning

Nama : Ali Akbar Bermano
NPM: G1A024041
Metode : Naive Bayes  
Dataset : diabetes.csv

import library

In [2]:
import pandas as pd
import numpy as np

from sklearn.model_selection import train_test_split
from sklearn.naive_bayes import GaussianNB
from sklearn.metrics import accuracy_score

Mount drive

In [5]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


Cek isi drive

In [6]:
!ls /content/drive/MyDrive

 16285991472532945197677496920095.jpg
 16285992569838900097044785282701.jpg
 16285993313501132899212429543543.jpg
 20241023092047000261-1729693286700.jpg
'Ali akbar bermano_G1A024041'
'ali akbar bermano_G1A024041.docx'
'Ali Akbar Bermano_ G1A024041_FormRegPKKFT24.pdf'
'ali akbar bermano_G1A024041.gdoc'
'Ali Akbar Bermano_G1A024041_KALKULUS.pptx'
'Ali akbar bermano_G1A024041_MSI-1.docx'
'Ali akbar bermano_G1A024041_MSI-1.pdf'
'Ali akbar bermano_G1A024041_RESUME JURNAL FISIKA (1).pdf'
'Ali Akbar Bermano_G1A024041_TI A.pdf'
'ALI AKBAR BERMANO (NPM: G1A024041) '
'[Bakadame.com] Overlord Volume 12 - The paladin of Holy Kingdom (1).pdf'
'[Bakadame.com] Overlord Volume 12 - The paladin of Holy Kingdom.pdf'
'[Bakadame.com] Overlord Volume 13 - The paladin of Holy Kingdom (1).pdf'
'[Bakadame.com] Overlord Volume 13 - The paladin of Holy Kingdom.pdf'
'-Bakadame.com- Overlord Volume 9 - The Magic Caster Of Destroy (1).pdf'
'-Bakadame.com- Overlord Volume 9 - The Magic Caster Of Destroy.pdf'
'Buku

load dataset

In [7]:
import pandas as pd

path = "/content/drive/MyDrive/diabetes.csv"

data = pd.read_csv(path)

data.head()

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,6,148,72,35,0,33.6,0.627,50,1
1,1,85,66,29,0,26.6,0.351,31,0
2,8,183,64,0,0,23.3,0.672,32,1
3,1,89,66,23,94,28.1,0.167,21,0
4,0,137,40,35,168,43.1,2.288,33,1


In [8]:
data.head()

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age,Outcome
0,6,148,72,35,0,33.6,0.627,50,1
1,1,85,66,29,0,26.6,0.351,31,0
2,8,183,64,0,0,23.3,0.672,32,1
3,1,89,66,23,94,28.1,0.167,21,0
4,0,137,40,35,168,43.1,2.288,33,1


eksplorasi data

In [9]:
data.shape

(768, 9)

In [10]:
data.columns

Index(['Pregnancies', 'Glucose', 'BloodPressure', 'SkinThickness', 'Insulin',
       'BMI', 'DiabetesPedigreeFunction', 'Age', 'Outcome'],
      dtype='object')

In [11]:
X = data.drop("Outcome", axis=1)
y = data["Outcome"]

In [12]:
data["Outcome"].value_counts()

,count
Outcome,
0,500
1,268


In [13]:
data["Outcome"].value_counts(normalize=True)

,proportion
Outcome,
0,0.651042
1,0.348958


Split Data (Membagi Data Training dan Testing)

In [14]:
from sklearn.model_selection import train_test_split

X_train, X_test, y_train, y_test = train_test_split(
    X,
    y,
    test_size=0.2,
    random_state=42
)

Training Model Naive Bayes

In [15]:
from sklearn.naive_bayes import GaussianNB

model = GaussianNB()

model.fit(X_train, y_train)

GaussianNB()

In [17]:
y_pred = model.predict(X_test)

menghitung mean

In [18]:
import pandas as pd

mean = X_train.groupby(y_train).mean()

mean

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age
Outcome,,,,,,,,
0,3.241895,110.214464,68.309227,19.748130,72.254364,30.256608,0.431254,30.556110
1,4.685446,140.887324,71.497653,21.624413,98.727700,35.234272,0.540545,37.333333


mengambil variance

In [19]:
variance = X_train.groupby(y_train).var()

variance

,Pregnancies,Glucose,BloodPressure,SkinThickness,Insulin,BMI,DiabetesPedigreeFunction,Age
Outcome,,,,,,,,
0,9.158840,678.328890,312.044140,212.618903,10233.220137,56.834362,0.094310,122.837469
1,13.093985,1070.185357,395.534193,285.301665,19298.010408,49.758584,0.142307,120.723270


mengambil data test

In [20]:
X_test.iloc[0]

,668
Pregnancies,6.00
Glucose,98.00
BloodPressure,58.00
SkinThickness,33.00
Insulin,190.00
BMI,34.00
DiabetesPedigreeFunction,0.43
Age,43.00


Accuracy Score / Evaluasi Model.

In [21]:
from sklearn.metrics import accuracy_score

accuracy = accuracy_score(y_test, y_pred)

print("Accuracy:", accuracy)

Accuracy: 0.7662337662337663


In [1]:
import pandas as pd

hasil = pd.DataFrame({
    "Actual": y_test,
    "Predicted": y_pred
})

hasil.head()

NameError: name 'y_test' is not defined

Model Naive Bayes digunakan untuk melakukan klasifikasi pada dataset diabetes.

Dataset dibagi menjadi data training dan testing dengan perbandingan 80:20.

Berdasarkan hasil evaluasi menggunakan accuracy score, model memiliki tingkat akurasi sekitar 0.77 atau 77%.

Hal ini menunjukkan bahwa model dapat memprediksi kemungkinan diabetes dengan cukup baik.